backend/infrahub/proposed_change/tasks.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from infrahub_sdk.protocols import CoreProposedChange
+from prefect import flow, task
+from prefect.logging import get_run_logger
+
+from infrahub.core.constants import ProposedChangeState
+from infrahub.services import (
+    services,
+)
+
+
+@flow(name="proposed-changes-cancel-branch", description="Cancel all Proposed change associated with a branch.")
+async def cancel_proposed_changes_branch(branch_name: str) -> None:
+    service = services.service
+    proposed_changed_opened = await service.client.filters(
+        kind=CoreProposedChange,
+        include=["id", "source_branch"],
+        state__value=ProposedChangeState.OPEN.value,
+        source_branch__value=branch_name,
+    )
+    proposed_changed_closed = await service.client.filters(
+        kind=CoreProposedChange,
+        include=["id", "source_branch"],
+        state__value=ProposedChangeState.CLOSED.value,
+        source_branch__value=branch_name,
+    )
+
+    for proposed_change in proposed_changed_opened + proposed_changed_closed:
+        await cancel_proposed_change(proposed_change=proposed_change)
+
+
+@task(description="Cancel a propose change")
+async def cancel_proposed_change(proposed_change: CoreProposedChange) -> None:
+    service = services.service
+    log = get_run_logger()
+
+    log.info("Canceling proposed change as the source branch was deleted")
+    proposed_change = await service.client.get(kind=CoreProposedChange, id=proposed_change.id)
```

Comment:
Not a big deal here and it's just copied code and something that won't happen that often. But it feels like we should have a way of doing this without having to get the node first and call the API twice when we know the type of node we want to update and we have the ID.